language/move-vm/runtime/src/data_cache.rs

```diff
@@ -42,7 +42,23 @@ pub trait RemoteCache {
         &self,
         address: &AccountAddress,
         tag: &StructTag,
-    ) -> PartialVMResult<Option<Vec<u8>>>;
+    ) -> PartialVMResult<Option<RefBytes>>;
+}
+
+pub enum RefBytes<'life> {
```

Comment:
is that a bootleg `Cow<'a, [u8]>` I see? : )
https://doc.rust-lang.org/std/borrow/enum.Cow.html

Reply:
ah I was trying to find something in std but ended up AsRef which requires 'static

Reply:
yeah, Cow is pretty cool! lets you return either an `Owned(Vec<u8>)` or a `Borrowed(&'a [u8])`. iirc there is also `supercow` (https://altsysrq.github.io/rustdoc/supercow/index.html), which `guppy` also uses.

Reply:
Should we just be using `Cow` here?

Reply:
I think we should use `Cow`.